src/witan/gateway/handler.clj

```diff
@@ -112,6 +112,26 @@
                          :kixi.comms.query/results results}))))
 
 ;;
+;;Add partition-key finder here!
+;;exception if there's isn't one
+
+(defn uuid
+  [& _]
+  (str (java.util.UUID/randomUUID)))
+
+(def command-key->partition-key-fn
+  {:kixi.datastore.filestore/create-upload-link uuid
+   :kixi.datastore.schema/create uuid
+   :kixi.datastore.filestore/create-file-metadata :kixi.datastore.metadatastore/id
+   :kixi.datastore.bundles/create-datapack :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/sharing-change :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/update :kixi.datastore.metadatastore/id})
+
+(defn partition-key-fn
+  [key]
+  (or (key command-key->partition-key-fn)
```

Comment:
command-key->partition-key-fn is a map, so this is abusing the "keyword as function" thing: (:a {:a 1). The 'key' arg name is shadowing the key fn, which is a bit pants really.